database/migrations/2024_12_03_174402_create_cities_table.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    public function up(): void
+    {
+        Schema::create('cities', function (Blueprint $table) {
+            $table->id();
+            $table->string('name')->unique();
```

Comment:
What if I have two cities, in different countries with the same name?

Reply:
We could solve the ambiguity by either adding:
- A new column `cities.country`.
- A new table `countries`. Then each city would reference a country.

For the scope of this project the first option would be sufficient. But, the second option would be more fitting in a production environment for scalability.

@Tincho44 What do you think?

Reply:
yeah, I feel it depends on the scope of the project. If it is something worldwide, maybe unique is not the way to go, there might be some country with repeated city name? But as we talk here in the office, if the requirement says there must not be repeated city name, you are totally right!

Reply:
Martin and I talked in the office about the city management requirements. According to execise:
> Input Field and Button to add a new city. Verify from the back and front that the city is not duplicated and that an empty string is not entered.

We then agreed that for the scope of the project the unique constraint is sufficient, but on a production environment, a `countries` table should be used